File: 2019-2020/2-january/silver/loan/loanf.py
import math
fin = open('loan.in', 'r')
fout = open('loan.out', 'w')
N,K,M = map(int, fin.readline().strip().split())
left = 1
right = int(K)
while right - left > 0:
    x = math.ceil((left+right)/2)
    k = math.ceil(math.log(M*x/N) / math.log((x-1)/x))
    if pow(x-1,k)* math.ceil(N/pow(x, k)) -M*(K-k) <= 0 :
        left = x
    else:
        right = x-1
fout.write(str(left) + "\n")

File: 2019-2020/2-january/silver/loan/loanf.cpp
#include <bits/stdc++.h>
using namespace std;
ofstream fout ("loan.out");
ifstream fin ("loan.in");

int main(){
    long long N,K,M;
    fin>>N>>K>>M;
    long long left = 1;
    long long right = K;
    while(right-left>0){
        long long x= ceil((left+right)/2.0);
        long long k = ceil(1.0* log(1.0*M*x/N) / log(1.0*(x-1)/x));
        long long cur = pow(x-1,k)*ceil(1.0*N/pow(x,k));
        if(cur - M*(K-k) <= 0){
            left = x;
        }
        else{
            right = x-1;
        }
    }
    fout<<left<<"\n";
}